iphone/cli/commands/_build.js

```diff
@@ -4892,1243 +5011,1266 @@ iOSBuilder.prototype.writeDebugProfilePlists = function writeDebugProfilePlists(
-					this.logger.trace(__('No change, skipping %s', info.dest.cyan));
-				}
-				this.unmarkBuildDirFile(info.dest);
-			}, this);
-		},
+	await this.writeAssetContentsFile(path.join(assetCatalogDir, 'Contents.json'), {
+		images: images,
+		info: {
+			version: 1,
+			author: 'xcode'
+		}
+	});
 
-		function (next) {
-			series(this, [
-				function processJSFiles(next) {
-					this.logger.info(__('Processing JavaScript files'));
-					const sdkCommonFolder = path.join(this.titaniumSdkPath, 'common', 'Resources', 'ios');
-					const task = new ProcessJsTask({
-						inputFiles: Object.keys(jsFiles).map(relPath => jsFiles[relPath].src),
-						incrementalDirectory: path.join(this.buildDir, 'incremental', 'process-js'),
-						logger: this.logger,
-						builder: this,
-						jsFiles,
-						jsBootstrapFiles,
-						sdkCommonFolder,
-						defaultAnalyzeOptions: {
-							minify: this.minifyJS,
-							transpile: this.transpile,
-							sourceMap: this.sourceMaps,
-							resourcesDir: this.xcodeAppDir,
-							logger: this.logger,
-							targets: {
-								ios: this.minIosVersion
-							}
-						}
-					});
-					task.run()
-						.then(() => {
-							this.tiSymbols = task.data.tiSymbols;
+	// If there are no missing launch logos, we're done
+	if (!missingLaunchLogos.length) {
+		return [];
+	}
 
-							return next(); // eslint-disable-line promise/no-callback-in-promise
-						})
-						.catch(e => {
-							this.logger.error(e);
-							process.exit(1);
-						});
-				},
+	if (!this.forceRebuild) {
+		this.logger.info(__('Forcing rebuild: launch logos changed since last build'));
+		this.forceRebuild = true;
+	}
 
-				function writeBootstrapJson() {
-					this.logger.info(__('Writing bootstrap json'));
+	if (!this.buildOnly && (this.target === 'device' || this.target === 'simulator')) {
+		this.logger.warn(__('If this app has been previously installed on this %s, you may need restart it to see the latest launch logo', this.target));
+		this.logger.warn(__('iOS renders and caches the launch screen to a PNG image that seems to only be invalidated by restarting iOS'));
+	}
 
-					const originalBootstrapJsonName = path.join('ti.internal', 'bootstrap.json');
-					const bootstrapJsonRelativePath = this.encryptJS ? path.join('ti_internal', 'bootstrap_json') : originalBootstrapJsonName;
-					const bootstrapJsonAbsolutePath = path.join(this.encryptJS ? this.buildAssetsDir : this.xcodeAppDir, bootstrapJsonRelativePath);
-					const bootstrapJsonString = JSON.stringify({ scripts: jsBootstrapFiles });
+	if (!launchLogo) {
+		// just use the DefaultIcon.png to generate the missing LaunchLogos
+		return missingLaunchLogos;
+	}
 
-					if (this.encryptJS) {
-						this.jsFilesEncrypted.push(originalBootstrapJsonName); // original name
-						this.jsFilesToEncrypt.push(bootstrapJsonRelativePath); // encrypted name
-					}
+	// Resize the launch logo to the other dimensions needed
+	await new Promise((resolve, reject) => {
```

Comment:
<!--
  0 failure: 
  1 warning:  iphone/cli/comman...
  
  
  DangerID: danger-id-Danger;
  File: iphone/cli/commands/_build.js;
  Line: 6102;
-->

- :warning: iphone/cli/commands/_build.js line 6102 – 'reject' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)